### DISCUSSION TITLE
How to visualise linear synteny between two de novo genome assemblies?

### DISCUSSION THREAD

**Discussion by siddharthjayaraman** [GMOD/jbrowse-components#1497] [Q&A] [answered]
Based on the documentation (https://www.jbrowse.org/jb2/docs/user_guide#linear-synteny-and-dotplot-views) it should be possible to view linear synteny using JBrowse2, but at the moment I am unable to load the config_synteny_grape_peach_paf.json in the test_data folder to test this feature set. 

The config_synteny_grape_peach.json in the same folder works but it doesn't use the *.paf alignment file. 

My question is, if I have two de novo assemblies ASM1 and ASM2 of a closely related species with their corresponding *.fas, *.fai and a ASM1_ASM2.paf (generated using Minimap2) files, what's the minimum configuration needed in the config.json to visualise the synteny? or do I need any additional datasets for it to work?

Thanks.

**Comment by cmdcolin** [comment, marked as answer]
@siddharthjayaraman I found there was a bug with config_synteny_grape_peach_paf.json (there was a recent overhaul to how synteny tracks are done to allow the same synteny track to be visible in both dotplot and linear modes)

I attempted a quick fix here

https://github.com/GMOD/jbrowse-components/pull/1498

Now config_synteny_grape_peach.json has an example with both MCScan and PAF file (I have tried to encourage PAF over MCScan generally because the MCScan is a little wild how it works, first having to fetch gene names from the mcscan, and then fetching the coordinates  from the gene track using subadapters....kinda crazy)

Glad to hear interest in this!

If you want to test it out locally you can


```
git clone https://github.com/gmod/jbrowse-components
cd jbrowse-components
yarn
cd products/jbrowse-web
yarn start
# nav to localhost:3000/?config=test_data/config_synteny_grape_peach.json
```

As far as loading your own instance, it is still something we are trying to work out properly. I think the best workflow is to use the GUI actually as opposed to the CLI. 

Using the GUI helps because it initializes the synteny view to contain the entire genome's scaffolds as displayedRegions

Note that order matters with the PAF file, it assumes the left column of the PAF is the top assembly, and the right column of the PAF is the bottom one... should probably document that

![localhost_3000__config=test_data%2Fconfig_synteny_grape_peach json (1) (1)](https://user-images.githubusercontent.com/6511937/100252756-3d559a80-2f0e-11eb-9e99-588f431fc9de.png)

**Reply by siddharthjayaraman** [reply]
Thanks for updating the example config json file. I was able to get it working and good to know about the column order in the paf file. 

Is there plan to introduce viewing linear synteny across multiple genomes, instead of just one pair?

**Reply by cmdcolin** [reply]
Yep it is true that we don't have this now but it should be possible in the future. This issue called multi-way synteny is tracking this issue https://github.com/GMOD/jbrowse-components/issues/1134
